modules/js-and-css/optimize-oembeds/hooks.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Hook callbacks used for oEmbed Optimizer.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$oembed_lazy_load_scripts = false;
+
+/**
+ * Filter the oEmbed HTML.
+ *
+ * Add loading="lazy" to any iframe tags.
+ * Lazy load any script tags.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $html The oEmbed HTML.
+ * @return string
+ */
+function perflab_optimize_oembed_html( $html ) {
+	global $oembed_lazy_load_scripts;
+
+	$p = new WP_HTML_Tag_Processor( $html );
+
+	// Find the first iframe or script tag to act on.
+	while ( $p->next_tag() ) {
+		if ( 'IFRAME' === $p->get_tag() ) {
+			$loading_value = $p->get_attribute( 'loading' );
+			if ( empty( $loading_value ) ) {
+				$p->set_attribute( 'loading', 'lazy' );
+			}
+			return $p->get_updated_html();
+
+		} elseif ( 'SCRIPT' === $p->get_tag() ) {
```

Comment:
This needs to make sure it only targets external scripts:

```suggestion
		} elseif ( 'SCRIPT' === $p->get_tag() && $p->get_attribute( 'src' ) ) {
```

Example oEmbed for Crowdsignal survey responses:

```html
<div class="pd-embed" data-settings="{&quot;type&quot;:&quot;iframe&quot;,&quot;auto&quot;:true,&quot;domain&quot;:&quot;rydk.survey.fm&quot;,&quot;id&quot;:&quot;test-survey&quot;}"></div>
<script type="text/javascript">
(function(d,c,j){if(!document.getElementById(j)){var pd=d.createElement(c),s;pd.id=j;pd.src=('https:'==document.location.protocol)?'https://polldaddy.com/survey.js':'http://i0.poll.fm/survey.js';s=document.getElementsByTagName(c)[0];s.parentNode.insertBefore(pd,s);}}(document,'script','pd-embed'));
</script>
```